hs-ucan/library/Web/UCAN/Validation.hs

```diff
@@ -181,3 +187,37 @@ checkEd25519Signature (UCAN.RawContent raw) ucan@UCAN {..} =
 
   where
     Claims {sender = DID.Key publicKey} = claims
+
+checkSecp256k1Signature ::
+     UCAN.RawContent
+  -> UCAN fct rsc ptc
+  -> Secp256k1.Signature
+  -> Either UCAN.Error (UCAN fct rsc ptc)
+checkSecp256k1Signature (UCAN.RawContent raw) ucan@UCAN {..} (Secp256k1.Signature innerSig) = do
+  case publicKey of
+    Secp256k1PublicKey pk ->
+      -- `innerSig` is actually a recoverable signature,
+      -- but since Ethereum (or Metamask?) often has a `v` value
```

Comment:
```suggestion
      -- but since Ethereum often has a `v` value
```

No need to mention Metamask here. Any Ethereum wallet(I would hope) would function in a consistent way 👍🏼  https://eips.ethereum.org/EIPS/eip-155

Reply:
Thanks, forgot to remove this! Regarding this, see the last paragraph in the PR description, would love your feedback on that 🙏

Reply:
> Thanks, forgot to remove this! Regarding this, see the last paragraph in the PR description, would love your feedback on that 🙏

@icidasset sure! I think [EIP-155](https://eips.ethereum.org/EIPS/eip-155) may be the key to distinguishing the two. as you mentioned, bitcoin and ethereum used to both have `v` values of 27(or sometimes 28?), but, looking at EIP-155, it says `v = CHAIN_ID * 2 + 35` or `v = CHAIN_ID * 2 + 36`, so maybe we can just check if `v >= 35`?

Reply:
Ah good point, I'll use `v >= 35`, thanks! Do you know of any other chains that prefix and/or hash the content before signing it?

Reply:
> Ah good point, I'll use `v >= 35`, thanks! Do you know of any other chains that prefix and/or hash the content before signing it?

hmm 🤔 I believe Tezos and Cosmos prefix them, but I'm not entirely sure how it works across most other chains(I've mostly worked in ethereum). I can do some digging though!

Reply:
That'd be great! Especially if we want to support other chains than Ethereum in the future 👀